### PR TITLE
Refactored Workflow model unique key to portal_run_id

### PR DESCRIPTION
**Pull request by victorskl**
i.e. Portal Automation business logic for tracking Workflow is
now centered around formula-driven `portal_run_id`, onwards.

* Added data migration to backfill historical portal_run_id null
* Altered wfr_id, wfl_id, wfv_id, version to nullable optional fields
* Dropped Workflow.sample_name field, related #244
* Added IdHelper.generate_portal_run_id()
* Removed redundant DragenWgtsQcWorkflowFactory mock
* Fixed related unit test cases due to refactoring

See https://umccr.slack.com/archives/CP356DDCH/p1693873519216669

### CODE REVIEW

**Comment by victorskl** [comment]
@alexiswl FYI

File: data_portal/fields.py [hunk 1]
```diff
@@ -45,3 +47,10 @@ def add(self, value):
 
     def calculate_hash(self):
         return self.__sha256.hexdigest()
+
+
+class IdHelper(object):
```
Comment (reisingerf) [inline data_portal/fields.py:52]: With the introduction of this `portal_run_id` generator function, should the `WorkflowHelper` not use this as well?
(now we have two places in which the IDs can be generated)

Comment (victorskl) [reply data_portal/fields.py:52]: Yup. That's need follow up there with changes in domain package. Plan is the `WorkflowHelper` needs to refactor to use this one.

**Review by reisingerf — APPROVED** [review]
LGTM

**Comment by victorskl** [comment]
Merging...

**Pull request merged**